File: app/src/main/java/ru/nikitaboiko/instagramgallery/MainActivity.java
package ru.nikitaboiko.instagramgallery;

import android.os.Bundle;

import androidx.appcompat.widget.SearchView;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.arellomobile.mvp.MvpAppCompatActivity;
import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.nikitaboiko.instagramgallery.adapters.GiphyAdapter;
import ru.nikitaboiko.instagramgallery.presenters.MainPresenter;

public class MainActivity extends MvpAppCompatActivity implements MainView {

    @BindView(R.id.activity_giphy_list)
    RecyclerView imageList;
    @BindView(R.id.activity_giphy_search_view)
    SearchView searchView;

    @Inject
    GiphyAdapter giphyAdapter;

    @InjectPresenter
    MainPresenter mainPresenter;
    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = item -> {
        switch (item.getItemId()) {
            case R.id.menu_giphy:
                return true;
            case R.id.menu_favourites:
                return true;
        }
        return false;
    };

    @ProvidePresenter
    public MainPresenter providePresenter() {
        MainPresenter presenter = new MainPresenter();
        App.instance().getAppComponent().inject(presenter);
        return presenter;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        App.instance().getAppComponent().inject(this);

        setContentView(R.layout.activity_main);
        BottomNavigationView navView = findViewById(R.id.nav_view);

        ButterKnife.bind(this);
        imageList.setLayoutManager(new LinearLayoutManager(this));
        imageList.setAdapter(giphyAdapter);
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {

            @Override
            public boolean onQueryTextSubmit(final String query) {
                setListCurrentSearchValue(query);
                return false;
            }

            @Override
            public boolean onQueryTextChange(final String newText) {
                return true;
            }
        });

        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }

    @Override
    public void setListCurrentSearchValue(String val) {
        mainPresenter.getGiphiesFromApi(val, giphyAdapter);
    }
}

File: app/src/main/java/com/example/instagramgallery/ui/activities/MainActivity.kt
package com.example.instagramgallery.ui.activities

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.example.instagramgallery.R
import com.example.instagramgallery.presenters.MainActivityPresenter
import com.example.instagramgallery.views.MainView
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : MvpAppCompatActivity(), MainView {

    @InjectPresenter
    internal lateinit var presenter: MainActivityPresenter

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.bottom_menu_gallery -> {
                presenter.changeFragment(1)
                return@OnNavigationItemSelectedListener true
            }
            R.id.bottom_menu_favourites -> {
                presenter.changeFragment(2)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val navView: BottomNavigationView = findViewById(R.id.nav_view)

        navView.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)

        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.container, presenter.galleryFragment)
        fragmentTransaction.commit()
    }

    override fun changeFragment(fragment: Fragment) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.container, fragment)
        fragmentTransaction.commit()
    }
}
